proxywasm/proxytest/root.go

```diff
@@ -276,12 +276,17 @@ func (r *rootHostEmulator) ProxyCallForeignFunction(funcNamePtr *byte, funcNameS
 	f, ok := r.foreignFunctions[funcName]
 	if !ok {
 		log.Fatalf("%s not registered as a foreign function", funcName)
+		return internal.StatusNotFound
+	}
+
+	ret, retval := f(param)
+
+	if len(ret) > 0 {
```

Comment:
I also added a fix for zero byte responses; this looked problematic as is.  Happy to split this to another PR if you like.